csrc/device_lower/validation.cpp

```diff
@@ -1290,4 +1290,26 @@ void validateResize(Fusion* fusion) {
   }
 }
 
+void validateReductions(Fusion* fusion) {
+  for (auto expr : fusion->exprs()) {
+    if (auto rop = dynamic_cast<ReductionOp*>(expr)) {
+      const std::vector<IterDomain*> in_rfactor = TensorDomain::noReductions(
+          rop->in()->as<TensorView>()->getMaybeRFactorDomain());
+      const std::vector<IterDomain*>& out_root =
+          rop->out()->as<TensorView>()->getRootDomain();
+
+      NVF_ERROR(in_rfactor.size() == out_root.size());
+      for (auto i : c10::irange(in_rfactor.size())) {
```

Comment:
Is it easier to use PairwiseRootDomainMap?

Reply:
Good advice. Thanks.